File: src/main/java/com/github/mslenc/fursinvoices/dto/FursResponse.java
package com.github.mslenc.fursinvoices.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class FursResponse {
    private String echoResponse;
    private InvoiceResponse invoiceResponse;
    private BusinessPremiseResponse businessPremiseResponse;

    @JsonProperty("EchoResponse")
    public String getEchoResponse() {
        return echoResponse;
    }

    @JsonProperty("EchoResponse")
    public void setEchoResponse(String echoResponse) {
        this.echoResponse = echoResponse;
    }

    @JsonProperty("InvoiceResponse")
    public InvoiceResponse getInvoiceResponse() {
        return invoiceResponse;
    }

    @JsonProperty("InvoiceResponse")
    public void setInvoiceResponse(InvoiceResponse invoiceResponse) {
        this.invoiceResponse = invoiceResponse;
    }

    @JsonProperty("BusinessPremiseResponse")
    public BusinessPremiseResponse getBusinessPremiseResponse() {
        return businessPremiseResponse;
    }

    @JsonProperty("BusinessPremiseResponse")
    public void setBusinessPremiseResponse(BusinessPremiseResponse businessPremiseResponse) {
        this.businessPremiseResponse = businessPremiseResponse;
    }
}

File: src/main/kotlin/com/github/mslenc/furslib/dto/FursResponse.kt
package com.github.mslenc.furslib.dto

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@JsonInclude(JsonInclude.Include.NON_EMPTY)
data class FursResponse(
    @JsonProperty("EchoResponse")
    val echoResponse: String? = null,

    @JsonProperty("InvoiceResponse")
    val invoiceResponse: InvoiceResponse? = null
)